crates/rustc_codegen_spirv/src/spirv_type.rs

```diff
@@ -440,6 +444,10 @@ impl fmt::Debug for SpirvTypePrinter<'_, '_> {
                 .field("access_qualifier", &access_qualifier)
                 .finish(),
             SpirvType::Sampler => f.debug_struct("Sampler").field("id", &self.id).finish(),
+            SpirvType::SampledImage { image_type } => f
+                .debug_struct("SampledImage")
+                .field("image_type", &self.cx.debug_type(image_type))
```

Comment:
Nit: this should have `.field("id", &self.id)` too (I've found including the id of self helps a lot when debugging type shenanigans)